pandas/core/internals/blocks.py

```diff
@@ -801,10 +801,23 @@ def _replace_list(
 
         rb = [self if inplace else self.copy()]
         for i, (src, dest) in enumerate(pairs):
+            convert = i == src_len  # only convert once at the end
             new_rb: List[Block] = []
+            mask_pos = 0
             for blk in rb:
-                m = masks[i]
-                convert = i == src_len  # only convert once at the end
+                if blk.ndim == 1:
+                    m = masks[i]
+                else:
+                    # GH-39338: _replace_coerce can split a block, so we
```

Comment:
if it gets split, will it be all-single-column?

Reply:
good question will look into that, would help simplify logic

Reply:
adjusted to account for this - split is happening in `split_and_operate`, which looks to always give single column result